### PR TITLE
Bl8 results component

### LINKED ISSUE #1143: Update search view to have icons on second line, no tooltips

### PR DESCRIPTION
**Pull request by dnoneill**
closes #1143 

Screenshot 2024-08-09 at 12 02 34 PM

### CODE REVIEW

File: app/assets/stylesheets/earthworks.css [hunk 1]
```diff
@@ -69,11 +69,7 @@
...
-  display: none;
-}
-
-.documents-list .documentHeader .index_title {
-  margin-left: .75rem;
-  align-self: center;
+.documents-list .document {
+  padding-top: 0px;
+  margin-top: 0px;
 }
```
Comment (thatbudakguy) [inline app/assets/stylesheets/earthworks.css:75]: would it work to override or append to `classes` in the component to add `pt-0 mt-0` so we don't need to do it here?

File: app/components/geoblacklight/header_icons_component.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Geoblacklight
+  class HeaderIconsComponent < ViewComponent::Base
```
Comment (thatbudakguy) [inline app/components/geoblacklight/header_icons_component.rb:4]: could we just make this `Earthworks::HeaderIconsComponent` too? I think it's easier to know when things are changed if they get namespaced that way

Comment (dnoneill) [reply app/components/geoblacklight/header_icons_component.rb:4]: Yeah I thought about that but then I think we need to pull out a bunch of components from geoblacklight and override those as well.

File: app/assets/stylesheets/earthworks.css (hunk 1 above)
Comment (dnoneill) [reply app/assets/stylesheets/earthworks.css:75]: I can check but I am pretty sure I would need to pull out a blacklight component to override the classes.

Comment (dnoneill) [reply app/assets/stylesheets/earthworks.css:75]: Yup. I would need to override the partial https://github.com/projectblacklight/blacklight/blob/a8b5a3d3a2e8243c8a875d40528eae5c106d83bd/app/views/catalog/_document_list.html.erb#L3 to do this.

Comment (dnoneill) [reply app/assets/stylesheets/earthworks.css:75]: Actually I might be wrong. Nevermind ignore me. Pretty sure this is doable.

File: app/components/geoblacklight/header_icons_component.rb (hunk 2 above)
Comment (thatbudakguy) [reply app/components/geoblacklight/header_icons_component.rb:4]: unless there's more places that aren't in that search, I think that is OK (and in fact good). the things in that search are:
- the component itself, which we are already changing
- a partial, which we would change to use the new component
- a test, which is not in earthworks
- a helper function, which we would change to use the new component
- the component that references this component, which is already changed in this PR

that seems like a pretty easy lift to me! just two places that aren't already in the PR, and neither is a component.

Comment (dnoneill) [reply app/components/geoblacklight/header_icons_component.rb:4]: I would be more inclined to change the way geoblacklight uses the component. Is there a reason we wouldn't want to update the config to set the headericon class in the controller?

Comment (thatbudakguy) [reply app/components/geoblacklight/header_icons_component.rb:4]: that is another path, for sure. from what i've seen that is mostly used for "big" stuff like the document component, sidebar, etc. but if that seems easier to you, go for it

Comment (dnoneill) [reply app/components/geoblacklight/header_icons_component.rb:4]: @thatbudakguy actually the helper I am pretty sure we don't want using our custom headericonscomponent. And the https://github.com/geoblacklight/geoblacklight/blob/9ce0dccdc2336dad520dfc0578743b20d421b0f6/app/views/catalog/_show_header_default.html.erb#L3 actually needs to get removed and fixed in geoblacklight (the document component missed this).

I am updating this PR.

**Review by thatbudakguy — APPROVED** [review]
lgtm, thanks!

**Pull request merged**